lib/internal/source_map/source_map_cache.js

```diff
@@ -148,17 +150,39 @@ function rekeySourceMap(cjsModuleInstance, newInstance) {
   }
 }
 
+// WARNING: The `sourceMapCacheToObject` and `appendCJSCache` run during
+// shutdown. In particular, they also run when Workers are terminated, making
+// it important that they do not call out to any user-provided code, including
+// built-in prototypes that might have been tampered with.
+const ObjectCreate = Object.create;
```

Comment:
these should already be tamper-proof because they're from `primordials`, are these saved here because of the performance issue bradley found?

Reply:
From https://github.com/orgs/nodejs/teams/collaborators/discussions/79, I’d say that yes, the goal is for us to move towards fully destructured primordials wherever possible.

(Personally, I also just like this for consistency.)

Reply:
cool. the way the comment is worded sounded like saving them here was what was providing the safety. just wanted to make sure everyone is on the same page :)